### ISSUE TITLE
Idea: Standardise the leave/done/depart text and hotkeys

### ISSUE THREAD

**Issue by brendanjones** [endless-sky/endless-sky#10000]
### Problem Description

As a seasoned player you'd think I'd be very used to these by now, but I still find myself hitting the wrong button to leave some places.

The various buttons we have to navigate away from places:

Done: Player Info, Map, Jobs (which is Map)
Leave: Shipyard, Outfitter
Depart: Leave planet
None: Messages (have to re-press the Messages hotkey, press escape, or click off the Messages panel), Spaceport (kinda related https://github.com/endless-sky/endless-sky/issues/3389), Bank, Trading, Hire Crew.

Pressing D will let you leave the Shipyard and Outfitter, but pressing L will not let you leave Player Info (it'll instead open the Logbook) or Map.

### Related Issue Links

Can't find any.

### Desired Solution

Is the idea that you don't accidentally leave the planet? Because then I'd say change `Leave` to `Done` in the Shipyard and Outfitter, and change `Depart` to `Leave`.

That standardises all dismissal actions to `Done` except for leaving the planet.

Messages should have a `Done` button added.

### Alternative Approaches

Open to discussion.

**Comment by brendanjones** [comment]
Hey Issue #10000 whaddupppppp that's a cool milestone.

**Comment by xX-Dillinger-Xx** [comment]
That all makes sense but I would like to suggest one small edit.

> and change `Depart` to `Leave`.  

Change `Depart` to `Launch`

**Comment by brendanjones** [comment]
> If I follow your idea correctly, `D` would exit everywhere except the planet and `L` would leave\launch. Is that correct? 

Correct.

> Change `Depart` to `Launch`

That is an excellent suggestion, will change the issue description.

**Comment by xX-Dillinger-Xx** [comment]
I really like this idea.

**Comment by Ferociousfeind** [comment]
Makes sense to me, especially if L is also Land, so L toggles whether or not you're on a planet 

Also happy 10k, we got there before the SCP wiki did

**Comment by brendanjones** [comment]
> especially if L is also Land, so L toggles whether or not you're on a planet

Yup, exactly. That was why I suggested Leave/Launch stays as L, while the others get changed to D.

**Comment by ziproot** [comment]
I think endless sky follows IBM CUA, so pressing `esc` lets you escape shipyard, outfitter, player info panel, message log, etc. The only time this won’t work is for departing a planet, which is the `d` key, as pressing `esc` takes you back to the main menu.

**Comment by TomGoodIdea** [comment]
Worth noting that <kbd>L</kbd> is currently used on the planet interface to return to the planet's description (same effect as clicking the landscape image).

**Comment by Ferociousfeind** [comment]
We should change it from "L" for Landscape to "D" for Describe :)

**Comment by brendanjones** [comment]
> (same effect as clicking the landscape image).

Had no idea that was a thing! Still no matter how long I play there are still things to learn :)

And yeah, if all the other non-Launch hotkeys are D then it seems logical to me to also switch that one to D.

**Comment by xobes** [comment]
I could do this.  Should I make a PR?

**Comment by Djorejk** [comment]
Can we have L and D share the same function on every screen, and find another key for Planet Description and Logbook?
Gives lefty and righty a conveniently reachable key for one of the most necessary actions.

**Comment by brendanjones** [comment]
The whole point of the changes I laid out is to have a different hotkey for leaving planet-side activities than for leaving the planet, so having all of them work with both L and D goes against that. In any case, we don't have left and right-handed versions of other hotkeys, why do we need that here? It's a game played with two hands.

**Comment by xobes** [comment]
> It's a game played with two hands.

Or a touchscreen... or a controller (perhaps?)...

If I'm understanding correctly, it's the departure that is the biggest problem...

Could add a dialog "you sure?" do the departure routine, and maybe shift-D could bypass the question (force departure)

The rest of this L/D doesn't bother me so much.  It's L to land.  Could be L to Launch, "are you sure"?, or shift-L to launch, no kidding.

Could perhaps add a cancelable count-down (3..2..1..Liftoff!) so that there's a recovery period... (esc. or C to cancel Launch)...

**Comment by Amazinite** [comment]
We're apprehensive about changing the hotkeys for buttons that have been this way for so long, because while changing them might help for you, that change could easily start confusing others who are used to the way that the buttons are now.

I'll note that the buttons on the planet panel can be customized in the interface file, so you could create a plugin that customizes these buttons. This is the button that returns you to the planet description:
https://github.com/endless-sky/endless-sky/blob/e49e1b2e53bf1ec16aea52036a7a0d22bceb7de1/data/_ui/interfaces.txt#L1429-L1431
And this is the button that launches you:
https://github.com/endless-sky/endless-sky/blob/e49e1b2e53bf1ec16aea52036a7a0d22bceb7de1/data/_ui/interfaces.txt#L1509-L1514

The shipyard and outfitter aren't currently customizable through the interface data at the moment, so you couldn't have a plugin change "Leave" to "Done", but that's something we can look into moving to the data to make it customizable.
